src/Graphics/CommandPool.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "Graphics/Vulkan/VulkanIncludes.h"
+#include "Graphics/Vulkan/VulkanExtensions.h"
+
+#include <map>
+
+namespace gore::gfx
+{
+
+class Device;
+
+class CommandPool
+{
+public:
+    CommandPool();
+    CommandPool(const Device& device, uint32_t queueFamilyIndex);
+    CommandPool(CommandPool&& other) noexcept;
+    ~CommandPool();
+
+    CommandPool& operator=(CommandPool&& other) noexcept;
+
+    [[nodiscard]] const vk::raii::CommandPool& Get(uint32_t swapchainImageIndex);
+
+    [[nodiscard]] const vk::raii::CommandBuffer& GetCommandBuffer(uint32_t swapchainImageIndex);
+    void Reset(uint32_t swapchainImageIndex);
+
+private:
+    struct Pool
```

Comment:
I'm using class Pool and class Handle to write graphics resource system. Could you please change the name of Pool even it is a private struct?

Reply:
Will do.